src/components/PlayerIcon/PlayerIcon.tsx

```diff
@@ -43,7 +43,7 @@ export function getPlayerIconURL(id, size): Promise<string> {{{
 
 export class PlayerIcon extends React.PureComponent<PlayerIconProps, {url}> {
     mounted: boolean = false;
-    listener;
```

Comment:
Would we need a `this.subscribe.to()` in `componentWillUnmount` here? Seems like this is a leak that exists with the current listener implementation too.